src/main/scala/akka/contrib/process/BlockingProcess.scala

```diff
@@ -107,16 +110,24 @@ class BlockingProcess(
     }
 
     try {
-      val stdin =
-        context.actorOf(OutputStreamSubscriber.props(process.getOutputStream), "stdin")
-
-      val stdout =
-        context.watch(context.actorOf(InputStreamPublisher.props(process.getInputStream, stdioTimeout), "stdout"))
-
-      val stderr =
-        context.watch(context.actorOf(InputStreamPublisher.props(process.getErrorStream, stdioTimeout), "stderr"))
-
-      context.parent ! Started(ActorSubscriber(stdin), ActorPublisher(stdout), ActorPublisher(stderr))
+      val stdin = Sink.outputStream(process.getOutputStream)
+        .mapMaterializedValue { futureResult =>
+          futureResult.onComplete(_ => context.stop(self))
```

Comment:
IOW If you can't actually watch the actors associated with stdio as we did before, then send a new message within your `onComplete` that is caught in the same way as `Destroy` and `Terminated`. Perhaps call it `StdioTerminated` or something.